### ISSUE TITLE
Spring Boot 2.0 Support

### ISSUE THREAD

**Issue by gbrehmer** [resilience4j/resilience4j#178]
I know Spring Boot 2 is not released yet, but most of the changes are already fixed (2.0.0.M6)
e.g. Endpoint configuration has changed significantly (AbstractEndpoint not available anymore). To support both versions, e.g. @ConditionalOnClass is required for configuration

**Comment by RobWin** [comment]
Would you like to contribute the changes?

**Comment by schosin** [comment]
Just as a heads-up, Spring Boot 2 also replaced Dropwizard with Micrometer. This might have an effect on resilience4j-metrics.

Looking forward to Spring Boot 2 support.

**Comment by HarishaAkasapu** [comment]
Seeing this issue after SpringBoot2 migration, any plans to address this?

**Comment by RobWin** [comment]
I started working on it.
Please see https://github.com/RobWin/resilience4j-spring-boot2-demo.
Add the dependencies as follows 
```
repositories {
	maven { url 'http://oss.jfrog.org/artifactory/oss-snapshot-local/' }
	mavenCentral()
}


ext{
	resilience4jVersion = '0.13.0-SNAPSHOT'
}

dependencies {
	compile('org.springframework.boot:spring-boot-starter-webflux')
	compile("io.github.resilience4j:resilience4j-spring-boot2:${resilience4jVersion}")
	compile("io.github.resilience4j:resilience4j-reactor:${resilience4jVersion}")
}
```

**Comment by HarishaAkasapu** [comment]
Thanks, will look into.

**Comment by radekkaminski** [comment]
How to apply the CircuitBreaker to Flux in this version ? CircuitBreakerOperator seems to be deleted...

**Comment by RobWin** [comment]
Please have a look at the demo on github. There us a link some comments above.
I guess you need resilience4j-reactor as a compile dependency.

**Comment by radekkaminski** [comment]
got it - thanks

**Comment by radekkaminski** [comment]
I can't connect Prometheus cause the simpleclient_spring_boot is dependant on Spring Boot 1.x - Does anyone know if there is an alternative ?

**Comment by RobWin** [comment]
Spring Boot2 uses MicroMeter which supports Prometheus out of the box.

**Comment by RobWin** [comment]
https://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#production-ready-metrics-export-prometheus

**Comment by michael-barker** [comment]
Any estimate when 0.13.0 will be released?  Looking forward to Spring Boot 2 support and some of the changes to Reactor support.

**Comment by RobWin** [comment]
Yes, the plan was to support it. But I didn't have time to implement it.
Would you like to create a PR?

**Comment by michael-barker** [comment]
Sure :)

**Comment by michael-barker** [comment]
How do I get an invite to the Slack channel?

**Comment by RobWin** [comment]
I need you email address. But then you would have to contribute even more PRs :) if that's your plan, I would love to invite you.

**Comment by michael-barker** [comment]
michael.barker.84 at gmail.  Thanks!

**Comment by Vity01** [comment]
Guys, could you already please release it? I need it immediately :-). I noticed it's in the git repo already.

**Comment by vijaycse** [comment]
Same here. Badly need the RecordFailurePredicates, RecordException and IgnoreException.

Can anyone tell me when its due?

**Comment by rLitto** [comment]
In review, hopefully tomorrow we could release

**Comment by RobWin** [comment]
v0.13.0 has been released

**Comment by Vity01** [comment]
:clap: Thank you.

**Comment by vijaycse** [comment]
@RobWin @Vity01  is the version released 0.13.1-SNAPSHOT or 0.13.1? I get 409 if i use 0.13.1 but snapshot works fine but not sure if snapshot is correct.

**Comment by rLitto** [comment]
0.13.0
